### ISSUE TITLE
Add warning when X or Y have redundant rows

### ISSUE THREAD

**Issue by ebridge2** [neurodata/hyppo#125]
When `X` or `Y` have redundant rows using `MGC`, the optimal scale will throw away redundant entries more or less (which makes sense), but this isn't immediately obvious reading through the output, and reduces the interpretability of the optimal scale. For example, if I have a perfectly linear relationship with two length 10 vectors `X` and `Y` where `xi = yi` for all `i`, I will get an optimal scale of `5` if there are `5` redundant entries. This may shift how I am otherwise interpreting the optimal scale returned. It may be advisable to either throw a warning telling people that they have redundant rows in `X`, `Y`, or both (and if `X` or `Y` are a distance matrix, this would translate to two elements with distances between them of exactly `0`) and remove these entries for them, or throw an error and tell people to do it themselves.

#### Reproducing code example:

```
from hyppo.independence import MGC as mgc_test
import numpy as np

X = np.hstack((np.arange(0, 6), 5, 5, 5, 5))
Y = np.hstack((np.arange(0, 6), 5, 5, 5, 5))

print(len(X))
>> 10
print(len(Y))
>> 10
```
and when we run `MGC`:
```
stat, pval, mgc_map = mgc_test().test(X, Y)
print(stat)
>> 1
print(pval)
>> .002
print(mgc_map["opt_scale"])
>> [6, 6]
```
If the user is not careful, they may interpret this to mean the relationship between `X` and `Y` is not linear, though that would be false since the optimal scale is not `[10, 10]` exclusively due to the redundancies. TL; DR is the upper bound for the `X` and `Y` scales (to my knowledge) will be the numbers of unique rows in each matrix (or the number of unique points which do not share a distance of zero between one another), and should probably be interpreted in that context.

#### Version information

- OS: Ubuntu 20.04
- Python Version 3.7.8
- Package Version 0.1.2

**Comment by Verathagnus** [comment]
This is my first time contributing to open source. I'm not sure if I understood the issue well enough. Please correct me.

**Comment by sampan501** [comment]
> This is my first time contributing to open source. I'm not sure if I understood the issue well enough. Please correct me.

The MGC algorithm returns a "map" of local correlations that give insights into the geometry of dependence in the relationship between inputs `X` and `Y`. The algorithm currently removes duplicate columns of this map. The optimal scale is the maximum value in this matrix, and when that value is in position `[n, n]`, the user generally knows that this relationship is linear. In the case of duplicate columns, even if the relationship is linear, the optimal scale will not be `[n, n]` (since that index doesn't exist), and so the user might think that the relationship must be nonlinear.

Basically, we need a warning given to the user in the case that there are values in `X` or `Y` that are repeated, so the user knows to take the fact the optimal scale is not equal to `[n, n]` with a grain of salt. Does that make sense?

**Comment by belkarx** [comment]
Is this issue still open? If so, may I work on it?

**Comment by sampan501** [comment]
not sure. @Verathagnus are you working on this?

**Comment by Verathagnus** [comment]
I linked this [pull request](https://github.com/Verathagnus/hyppo/pull/1) for trying to solve the issue. Is there any issue with the code?

**Comment by PSSF23** [comment]
@Verathagnus you need to create a PR to the `main` branch here instead of your own fork...

@sampan501 how do you think of the warnings?
